src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -339,11 +336,22 @@ public function reset()
      *
      * @return array An array of all defined service ids
      */
-    public function getServiceIds()
+    public function getServiceIds(/*$includePrivates = true*/)
     {
+        $includePrivates = true;
+        if (func_num_args() === 1) {
+            $includePrivates = !!func_get_arg(0);
+            if ($includePrivates) {
+                @trigger_error(sprintf('Including private services in the list of service id\'s is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id), E_USER_DEPRECATED);
```

Comment:
Sorry, I don't understand your concerns 😕 ?

You can't just deprecate privates in `getServiceIds` if you have no idea the privates were asked. And simply not returning them will be a BC break. Thus, the new argument to explicit the intention.